### PR TITLE
Update GH Actions workflow dependencies

### PR DESCRIPTION
**Pull request by nmoroze**
A lot of our Actions dependencies were pinned to old versions that use deprecated features. Updating them should silence the warnings we see in our workflow summary pages.

### CODE REVIEW

**Comment by nmoroze** [comment]
Should probably hold off on merge till this goes green: https://github.com/siliconcompiler/siliconcompiler/actions/runs/3943957889.

I'm not too worried about trying daily tests before merging, that's a subset of the quick test changes and it will bog down the test runner for a while.

**Comment by gadfort** [comment]
@nmoroze I spoke with @aolofsson and I think it will be easiest to fix the wheels once everything else is done, it's failing on some tests that require openroad, so to avoid repeating the same fixes over and over again, I suggested leaving that one red (daily CI and push CI should still stay green). And once the bulk of the changes are one we can fix the wheels.
Let me know if you think we should do otherwise.

**Comment by nmoroze** [comment]
Whoops, I just went in and fixed it before noticing your reply. I think if this works we might as well merge, but otherwise I'll leave them alone until after the dust settles.

**Review by gadfort — APPROVED** [review]
Lgtm

**Pull request merged**
